.github/workflows/build-and-run-gpu.yml

```
@@ -0,0 +1,127 @@
+name: Build-And-Run Tests on GPU
+
+on:
+  pull_request:
+    types: [ labeled, opened, synchronize, reopened ]
```

Let's add a comment here that clarifies that the default `types` are `opened`, `synchronize`, and `reopened`, to which we are adding `labeled`.

https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request

```suggestion
    # Trigger the workflow whenever a PR receives a special label
    # in addition to the default triggering events.
    types: [ labeled, opened, synchronize, reopened ]
```